dace/codegen/compiled_sdfg.py

```diff
@@ -503,39 +576,39 @@ def _construct_args(self, kwargs) -> Tuple[Tuple[Any], Tuple[Any]]:
             # Null pointer
             elif arg is None and isinstance(argtype, dt.Array):
                 arglist[index] = ctypes.c_void_p(0)
+            #
 
-        # Retain only the element datatype for upcoming checks and casts
-        arg_ctypes = [t.dtype.as_ctypes() for t in argtypes]
-
-        sdfg = self._sdfg
+            # Retain only the element datatype for upcoming checks and casts
+            arg_ctypes.append( argtypes[index].dtype.as_ctypes() )
+        #
+        return arg_ctypes
 
-        # Obtain SDFG constants
-        constants = sdfg.constants
 
-        # Remove symbolic constants from arguments
-        callparams = tuple((arg, actype, atype, aname)
-                           for arg, actype, atype, aname in zip(arglist, arg_ctypes, argtypes, argnames)
-                           if not symbolic.issymbolic(arg) or (hasattr(arg, 'name') and arg.name not in constants))
+    def _construct_args_callparams(self, arglist, arg_ctypes, argtypes, argnames):
+        constants  = self.sdfg.constants
+        callparams = []
```

Comment:
List creation and dynamically expanding its size can be expensive, performance wise.